src/ObjectUtils/PdbFileLlvm.cpp

```diff
@@ -31,41 +28,38 @@ namespace {
 // all symbol info required for functions.
 class SymbolInfoVisitor : public llvm::codeview::SymbolVisitorCallbacks {
  public:
-  SymbolInfoVisitor(ModuleSymbols* module_symbols, const ObjectFileInfo& object_file_info,
+  SymbolInfoVisitor(DebugSymbols* debug_symbols, const ObjectFileInfo& object_file_info,
                     llvm::pdb::TpiStream* type_info_stream)
-      : module_symbols_(module_symbols),
+      : debug_symbols_(debug_symbols),
         object_file_info_(object_file_info),
         type_info_stream_(type_info_stream) {
-    ORBIT_CHECK(module_symbols != nullptr);
+    ORBIT_CHECK(debug_symbols != nullptr);
     ORBIT_CHECK(type_info_stream != nullptr);
   }
 
   // This is the only record type (ProcSym) we are interested in, so we only override this
   // method. Other records will simply return llvm::Error::success without any work done.
   llvm::Error visitKnownRecord(llvm::codeview::CVSymbol& /*unused*/,
                                llvm::codeview::ProcSym& proc) override {
-    SymbolInfo symbol_info;
-    symbol_info.set_name(proc.Name.str());
-    symbol_info.set_demangled_name(llvm::demangle(proc.Name.str()));
+    FunctionSymbol& function_symbol = debug_symbols_->function_symbols.emplace_back();
+    function_symbol.name = proc.Name.str();
 
     // The ProcSym's name does not contain an argument list. However, this information is required
     // when dealing with overloads and it is available in the type info stream. See:
     // https://llvm.org/docs/PDB/TpiStream.html
     llvm::StringRef argument_list = RetrieveArgumentList(proc);
     if (!argument_list.empty()) {
-      symbol_info.set_demangled_name(
-          absl::StrCat(symbol_info.demangled_name(), argument_list.data()));
+      function_symbol.argument_list = argument_list.data();
```

Comment:
As stated above, I think we really should not store the argument list as an extra field. For PDB files, the name should be always demangled, so we can just remove the call to llvm::demangle here.